### ISSUE TITLE
Multiple approved comments depending on the number of labels.

### ISSUE THREAD

**Issue by wadakatu** [dkhmelenko/autoapproval#596]
Hello.
I have a problem with multiple approved comments from your bot.

## Problem

I got multiple approved comments from autoapproval bot depending on the number of labels.

If I put three labels when I create a pull request, I got three approved comments.
If I put five labels when I create a pull request, I got five approved comments.

After I create a pull request, the change of labels does not trigger sending approved comments.

## autoapproval.yml

This is my autoapproval.yml.

```yml
from_owner:
  - wadakatu
  - dependabot[bot]
required_labels: []
apply_labels: []
```

## Evidence

https://github.com/wadakatu/Koyolympus/pull/129
https://github.com/wadakatu/Koyolympus/pull/130
https://github.com/wadakatu/Koyolympus/pull/131

Could you tell me this behavior is as you expected or not?
If you need more information, please let me know.
Thank you.

**Comment by dabrady** [comment]
I was just evaluating whether this app suited my needs and happened to see your issue @wadakatu. At first glance, I would assume that behavior is not intended, but I understand why it happens:
1. When you are preparing a PR (before creating it) you have the option to do things like add labels and request reviewers and such.
2. When you do this, and the PR is created, GitHub fires multiple events at the same time: the `pull_request.opened` event, plus a `pull_request.labeled` event for each label you specified during creation.
3. This GitHub App is triggered by both a `pull_request.opened` event and a `pull_request.labeled` event and thus would be triggered multiple times when opening the PR: https://github.com/dkhmelenko/autoapproval/blob/d85d687475e395e55fac70532aa1a2d880c45a1e/src/index.ts#L5
4. Your app configuration doesn't care about labels, though, just the author, so each of those app runs would satisfy the criteria for auto-approval, resulting in multiple attempts at approving.

I do not know the reason this app is hardcoded to respond to labeling events, but without a code change to the app source, **the only workaround I can think of is to avoid adding labels until after the PR has opened** and triggered this app the first time: reading the source a bit deeper, you can see that once the PR already has an approval, the app won't try to approve again. https://github.com/dkhmelenko/autoapproval/blob/d85d687475e395e55fac70532aa1a2d880c45a1e/src/index.ts#L45-L48
(cc @dkhmelenko)

**Comment by wadakatu** [comment]
@dabrady 
Thank you for your detailed comment.
I could fully understand why it happened.

It seems like `pull_request.labeled` is not necessary as one of the arguments of app.on().

I would love to hear the author's opinion about it.

**Comment by dkhmelenko** [comment]
Thanks for reporting this issue. It looks like GitHub changed the way the events are fired. In the past, the event `pull_request.labeled` was not raised right after the `pull_request.opened`. This issue was not the case in the past, but now it is.

I am not sure how annoying is it for you. I will be able to look into it later and will try to fix it. Meanwhile, feel free to create a PR.

**Comment by dabrady** [comment]
(By the way, I think #484 shares the same root cause as this issue)

**Comment by dkhmelenko** [comment]
I also thought about it. But in that case, it seems that the bot gets stuck somehow and infinitely approves PRs. I was not able to reproduce it though.

**Comment by dabrady** [comment]
@dkhmelenko @wadakatu I've submitted a PR to fix this bug: #604

**Comment by dkhmelenko** [comment]
Fixed in https://github.com/dkhmelenko/autoapproval/pull/604
